GVRf/Extensions/ResonanceAudio/resonanceaudio/src/main/java/org/gearvrf/resonanceaudio/GVRAudioManager.java

```diff
@@ -0,0 +1,240 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.resonanceaudio;
+
+import com.google.vr.sdk.audio.GvrAudioEngine;
+
+import org.gearvrf.GVRActivity;
+import org.gearvrf.GVRCameraRig;
+import org.gearvrf.GVRContext;
+import org.gearvrf.GVRDrawFrameListener;
+import org.gearvrf.GVREventListeners;
+import org.gearvrf.GVRTransform;
+import org.joml.Quaternionf;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Manages audio sources for the applications.
+ * Handles pausing and resuming sound and maintains
+ * a list of {@link GVRAudioSource} objects which are attached
+ * to scene objects. The position of each audio source is
+ * obtained from the {@link GVRTransform} of the {@GVRSceneObject}
+ * the audio source is attached to. The audio manager also
+ * tracks the head transform so the sound will be spatially
+ * correct.
+ */
+public class GVRAudioManager extends GVREventListeners.ActivityEvents
```

Comment:
Extending ActivityEvents and implementing GVRDrawFrameListener pollutes the api space with unnecessary methods. Could as easily be implemented as inner classes.